internal/queue/loops/looper.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright (c) 2025 Diagrid Inc.
+Licensed under the MIT License.
+*/
+
+package loops
+
+import (
+	"context"
+	"sync"
+)
+
+type Handler[T any] interface {
+	Handle(context.Context, T) error
+}
+
+type Options[T any] struct {
+	Handler    Handler[T]
+	BufferSize *uint64
+}
+
+type Interface[T any] interface {
+	Run(ctx context.Context) error
+	Enqueue(req T)
+	Close(req T)
+}
+
+type looper[T any] struct {
+	queue   chan T
+	handler Handler[T]
+
+	closed  bool
```

Comment:
Should we use an `atomic.Bool` here?
```suggestion
	closed  atomic.Bool
```

Reply:
No, closed is only changed inside a lock so is superfluous and not required.